pkg/odo/cli/component/common_push.go

```diff
@@ -31,7 +31,7 @@ type CommonPushOptions struct {
 
 	sourceType       config.SrcType
 	sourcePath       string
-	componentContext string
+	ComponentContext string
```

Comment:
@mik-dass are we comfortable making this an exported element of the struct instead of private?

Reply:
This will allow the other packages to modify the variable.

Reply:
I understand that. And that's why I was wondering if we are comfortable with it.

Reply:
I don't think we should go ahead with this approach.

Reply:
@niconosenzo there's a suggestion on what approach might be helpful on the issue https://github.com/openshift/odo/issues/4340#issuecomment-775921042. Do you think you want to continue working on this? 

It's fine if you'd not feel comfortable doing so. There are plenty of other issue on the repo to pick from.